vizro-core/tests/unit/vizro/models/_components/form/test_date_picker.py

```diff
@@ -0,0 +1,175 @@
+    def test_valid_title(self, title):
+        date_picker = vm.DatePicker(title=title)
+
+        assert date_picker.title == str(title)
+
+    def test_set_action_via_validator(self, identity_action_function):
+        date_picker = vm.DatePicker(actions=[vm.Action(function=identity_action_function())])
+        actions_chain = date_picker.actions[0]
+
+        assert actions_chain.trigger.component_property == "value"
+
+    @pytest.mark.parametrize("min, max", [("2024-01-01", None), (None, "2024-01-01"), ("2024-01-01", "2024-02-01")])
+    def test_valid_min_max(self, min, max):
+        date_picker = vm.DatePicker(min=min, max=max)
+
+        assert date_picker.min == (datetime.strptime(min, "%Y-%m-%d").date() if min else None)
+        assert date_picker.max == (datetime.strptime(max, "%Y-%m-%d").date() if max else None)
+
+    def test_validate_max_invalid_min_greater_than_max(self):
+        with pytest.raises(
+            ValidationError, match="Maximum value of component is required to be larger than minimum value."
+        ):
+            vm.DatePicker(min="2024-02-01", max="2024-01-01")
+
+    def test_validate_max_invalid_date_format(self):
+        with pytest.raises(ValidationError, match="invalid date format"):
+            vm.DatePicker(min="50-50-50", max="50-50-50")
+
+    @pytest.mark.parametrize(
+        "range, value",
+        [
+            (False, "2024-01-01"),
+            (False, date(2024, 1, 1)),
+            (True, ["2024-01-01", "2024-02-01"]),
+            (True, [date(2024, 1, 1), date(2024, 2, 1)]),
+        ],
+    )
+    def test_validate_datepicker_value_valid(self, range, value):
+        date_picker = vm.DatePicker(min="2024-01-01", max="2024-02-01", range=range, value=value)
+
+        value_to_date = pd.to_datetime(value).to_pydatetime()
+        if isinstance(value_to_date, Collection):
+            value_to_date = [v.date() for v in value_to_date]
+        else:
+            value_to_date = value_to_date.date()
+
+        assert date_picker.value == value_to_date
+
+    @pytest.mark.parametrize(
+        "range, value",
+        [
+            (False, "2024-01-01"),
+            (False, date(2024, 1, 1)),
+            (True, ["2024-01-01", "2024-02-01"]),
+            (True, [date(2024, 1, 1), date(2024, 2, 1)]),
+        ],
+    )
+    def test_validate_datepicker_value_invalid(self, range, value):
+        with pytest.raises(ValidationError, match="Please provide a valid value between the min and max value."):
+            vm.DatePicker(min="2023-01-01", max="2023-02-01", range=range, value=value)
+
+    @pytest.mark.parametrize(
+        "range, value",
+        [
+            (False, "2024-01-01"),
+            (False, date(2024, 1, 1)),
+            (True, ["2024-01-01", "2024-02-01"]),
+            (True, [date(2024, 1, 1), date(2024, 2, 1)]),
+        ],
+    )
+    def test_validate_datepicker_range_valid(self, range, value):
```

Comment:
Similar comment here about why we want to test both a string and a `date` object.